api/src/hyp3_api/handlers.py

```diff
@@ -102,6 +114,23 @@ def get_request_time_expression(start, end):
         return key.lte(formatted_end)
 
 
+def check_granules_exists(granules):
+    cmr_parameters = {
+        'producer_granule_id': granules,
+        'provider': 'ASF',
+        'collection_concept_ids': {
+            "C1214470488-ASF",  # SENTINEL-1A_SLC
+            "C1327985661-ASF",  # SENTINEL-1B_SLC
+        }
+    }
+    response = requests.get('https://cmr.earthdata.nasa.gov/search/granules.json', params=cmr_parameters)
+    response.raise_for_status()
+    found_granules = [entry['producer_granule_id'] for entry in response.json()['feed']['entry']]
+    if set(granules) != set(found_granules):
```

Comment:
you can use the subtraction operator on sets to the the difference, https://docs.python.org/2/library/sets.html#set-objects
```
not_found_granules = granules - found_granules
if not_found_granules:
    raise CmrError(f'Requested scenes could not be found: {",".join(not_found_granules)}')
```